modules/axtask/src/task.rs

```diff
@@ -58,7 +58,7 @@ pub struct TaskInner {
     on_cpu: AtomicBool,
     /// A weak reference to the previous task running on this CPU.
     #[cfg(feature = "smp")]
-    prev_task: UnsafeCell<Weak<AxTask>>,
+    prev_task: UnsafeCell<Option<AxTaskRef>>,
```

Comment:
@equation314 please mind here, I use `Arc` instead of `Weak`, because we need a place to store the reference to "migrated task" which is not put into the run queue
details are explained in https://github.com/arceos-org/arceos/issues/197#issuecomment-2452913756

Reply:
I think we should still use the `Weak` pointer to save `prev_task`, but keep an additional per-CPU variable `migrating_task` to avoid affecting the reference count of normal tasks.

BTW, I found that `prev_task` can also be a per-CPU variable instead of a per-task variable to simplify the `TaskInner` structure.

Reply:
> I think we should still use the `Weak` pointer to save `prev_task`, but keep an additional per-CPU variable `migrating_task` to avoid affecting the reference count of normal tasks.

The `Weak` pointer for `prev_task` is just used to finished the scheduling process of prev task by setting its `on_cpu` flag as false.

> BTW, I found that prev_task can also be a per-CPU variable instead of a per-task variable to simplify the TaskInner structure.

if `prev_task` is stored in a per-CPU variable, yes, we can simply put all operations related with prev task after `resched`, just put them in `finish_prev_task_switch`, including selecting a run queue for prev task due to normal yielding